File: mocha-game/src/main/java/mocha/game/world/item/itemprototype/ItemPrototype.java
package mocha.game.world.item.itemprototype;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import mocha.game.world.item.ItemType;
import mocha.shared.Identified;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class ItemPrototype implements Identified<Integer> {

  private Integer id;
  private String name;
  private String spriteId;
  private ItemType itemType;
  private String description;

}

File: mocha-game/src/main/java/mocha/game/world/item/itemprototype/ItemPrototype.kt
package mocha.game.world.item.itemprototype

import mocha.game.world.item.ItemType
import mocha.shared.Identified

open class ItemPrototype : Identified<Int> {

  private var id: Int? = null
  open var name: String? = null
  open var spriteId: String? = null
  open var itemType: ItemType? = null
  open var description: String? = null

  constructor()

  constructor(id: Int?, name: String, spriteId: String, itemType: ItemType, description: String) {
    this.id = id
    this.name = name
    this.spriteId = spriteId
    this.itemType = itemType
    this.description = description
  }

  override fun getId(): Int? {
    return id
  }

  override fun setId(id: Int?) {
    this.id = id
  }
}
